File: examples/reactNative-demo/android/app/src/main/java/com/reactnativedemo/plaud/PlaudUploadModule.kt
package com.reactnativedemo.plaud

import android.util.Log
import com.facebook.react.bridge.*
import com.facebook.react.modules.core.DeviceEventManagerModule

class PlaudUploadModule(reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    private val TAG = "PlaudUploadModule"

    override fun getName(): String {
        return "PlaudUpload"
    }

    /**
     * Send event to React Native
     */
    private fun sendEvent(eventName: String, params: WritableMap?) {
        reactApplicationContext
            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
            .emit(eventName, params)
    }

    /**
     * Upload file to cloud
     */
    @ReactMethod
    fun uploadFile(filePath: String, options: ReadableMap?, promise: Promise) {
        try {
            Log.d(TAG, "Uploading file: $filePath")
            
            // TODO: Implement file upload logic
            
            promise.resolve(Arguments.createMap().apply {
                putBoolean("success", true)
                putString("filePath", filePath)
                putString("message", "File uploaded successfully")
            })

        } catch (e: Exception) {
            Log.e(TAG, "Upload file error", e)
            promise.reject("UPLOAD_ERROR", "Upload file error: ${e.message}", e)
        }
    }

    /**
     * Add event listener method
     */
    @ReactMethod
    fun addListener(eventName: String) {
        // React Native needs this method to avoid warnings
    }

    /**
     * Remove event listener method
     */
    @ReactMethod
    fun removeListeners(count: Int) {
        // React Native needs this method to avoid warnings
    }
}

File: examples/reactNative-demo/ios/PlaudSDK/PlaudUploadModule.m
//
//  PlaudUploadModule.m
//  ReactNativeDemo
//
//  Created by Plaud Team on 2024/12/19.
//

#import "PlaudUploadModule.h"
#import <React/RCTLog.h>

@implementation PlaudUploadModule

RCT_EXPORT_MODULE(PlaudUpload);

- (NSArray<NSString *> *)supportedEvents {
    return @[@"onUploadProgress", @"onUploadComplete", @"onUploadError"];
}

RCT_EXPORT_METHOD(uploadFile:(NSString *)filePath
                  options:(NSDictionary *)options
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
{
    RCTLogInfo(@"[iOS] PlaudUploadModule uploadFile called with path: %@", filePath);
    
    @try {
        // TODO: Implement actual file upload logic
        // For now, simulate success
        
        NSDictionary *result = @{
            @"success": @YES,
            @"filePath": filePath,
            @"message": @"File uploaded successfully (iOS placeholder)"
        };
        
        resolve(result);
        
        // Send completion event
        [self sendEventWithName:@"onUploadComplete" body:@{
            @"filePath": filePath,
            @"success": @YES
        }];
        
    } @catch (NSException *exception) {
        RCTLogError(@"[iOS] PlaudUploadModule uploadFile error: %@", exception.reason);
        reject(@"UPLOAD_ERROR", [NSString stringWithFormat:@"Upload file error: %@", exception.reason], nil);
    }
}

RCT_EXPORT_METHOD(addListener:(NSString *)eventName)
{
    // Required for RCTEventEmitter
}

RCT_EXPORT_METHOD(removeListeners:(NSInteger)count)
{
    // Required for RCTEventEmitter
}

@end
